### PR TITLE
DB-11687 Improve NE+Range estimates

### PR DESCRIPTION
**Pull request by ascend1**
## Short Description
This change boost the performance of query with hash `SGgiYFr5`.

## Long Description
Three issues:
* Index XPFVEZUASERLZP is not picked up probably because high number of estimated output rows. Currently working on this.
* Join between VE and VZ is not NLJ probably due to the same issue above.
* We don’t have bitmap IXAND.

To improve the first two points, this change improves the following:
* A not equal predicate and a range predicate are combined if the not equal value equals to an inclusive boundary value of the range predicate.

## How to test
`SGgiYFr5` should have plan on OLTP after this change. It's execution time should below 30 seconds on a k8s cluster. Before this change, the plan is on OLAP and takes at least 350 seconds.

### CODE REVIEW

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by msirek — APPROVED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/NestedLoopJoinStrategy.java [hunk 1]
```diff
@@ -367,53 +367,15 @@ private boolean hasJoinPredicateWithIndexKeyLookup(OptimizablePredicateList pred
      */
 
     public static double nestedLoopJoinStrategyLocalCost(CostEstimate innerCost, CostEstimate outerCost,
-                                                         double numOfJoinedRows, boolean useSparkCostFormula) {
+                                                         double numOfJoinedRows) {
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/NestedLoopJoinStrategy.java:370]: I think this is dangerous. I'm not too happy about the different formulas for control and Spark, but the current Spark penalty only applies in some cases, so we could end up choosing NLJ in Spark where it would be very costly.

**Review by dgomezferro — CHANGES_REQUESTED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/NestedLoopJoinStrategy.java (hunk 1 above)
Comment (ascend1) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/NestedLoopJoinStrategy.java:370]: Right. I thought about it yesterday again for some time and I think it makes sense to revert back to two formulae. One reason is to avoid regression on OLAP, the other reason is that we probably do have special treatments for OLTP later.

Comment (ascend1) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/compile/NestedLoopJoinStrategy.java]: Reverted and rebased.

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by carolp-503 — APPROVED** [review]
+1

**Review by dgomezferro — APPROVED** [review]
